server/src/main/java/org/elasticsearch/index/engine/Segment.java

```diff
@@ -168,7 +168,7 @@ public void writeTo(StreamOutput out) throws IOException {
         boolean hasAttributes = attributes != null;
         out.writeBoolean(hasAttributes);
         if (hasAttributes) {
-            out.writeMap(attributes, StreamOutput::writeString);
+            out.writeMap(attributes, StreamOutput::writeGenericValue);
```

Comment:
Ah, we cannot do this in a BWC way. I think we should still strive for it to be in attributes as `String, String`. 

So, either its in attributes as string string `Format: "comma,delimited,field,names"` or as new fields.

I don't think we need the `codec` itself, just the knn field formats.

Reply:
how about something like this ? 

```json
{
  "_shards": {
    "total": 2,
    "successful": 1,
    "failed": 0
  },
  "indices": {
    "collection-with-embeddings": {
      "shards": {
        "0": [
          {
            "routing": {
              "state": "STARTED",
              "primary": true,
              "node": "pbNCRypvR7KWdgrmAdvFuw"
            },
            "num_committed_segments": 0,
            "num_search_segments": 1,
            "segments": {
              "_0": {
                "generation": 0,
                "num_docs": 100,
                "deleted_docs": 0,
                "size_in_bytes": 579816,
                "committed": false,
                "search": true,
                "version": "9.10.0",
                "compound": true,
                "attributes": {
-->            "ES813FlatVectorFormat": "[text_embedding]",
                  "Zstd814StoredFieldsFormat.mode": "BEST_SPEED"
                }
              }
            }
          }
        ]
      }
    }
  }
}

```

Reply:
Yes, I think that will work. Could you add some tests ensuring we return expected values? I am not sure these could easily be `yaml` tests due to the format version changes that can happen from version to version. But having some internal integration test or something would be good.